src/errors.rs

```diff
@@ -1,74 +1,102 @@
     #[error("error while reading response: [{0}]")]
     PayloadRecvError(reqwest::Error),
+    /// An error returned when the base response structure is malformed e.g. meta properties like rpc version are missing.
     #[error("error while parsing server response: [{0:?}]")]
     PayloadParseError(message::Broken),
+    /// An error returned when the internal response structure is malformed, e.g. block data in a block method response is missing.
     #[error(transparent)]
     ResponseParseError(JsonRpcTransportHandlerResponseError),
 }
 
+/// Errors returned while sending requests to or receiving responses from the rpc server.
 #[derive(Debug, Error)]
 pub enum RpcTransportError {
+    /// Errors returned while sending a request to the rpc node.
     #[error(transparent)]
     SendError(JsonRpcTransportSendError),
+    /// Errors returned while receiving a response from an rpc server.
     #[error(transparent)]
     RecvError(JsonRpcTransportRecvError),
 }
 
+/// Meta errors returned by the rpc server.
 #[derive(Debug, Error)]
 pub enum JsonRpcServerResponseStatusError {
+    /// An error returned when the rpc client is unauthorized.
     #[error("this client is unauthorized")]
     Unauthorized,
+    /// An error returned when the rpc client exceeds the rate limit by sending too many requests.
     #[error("this client has exceeded the rate limit")]
     TooManyRequests,
+    /// An error returned when the rpc server returns a non-200 status code.
     #[error("the server returned a non-OK (200) status code: [{status}]")]
     Unexpected { status: reqwest::StatusCode },
 }
 
+/// Errors returned by the rpc server.
 #[derive(Debug, Error)]
 pub enum JsonRpcServerError<E> {
+    /// An error returned when the rpc server is unable to validation a client's request.
     #[error("request validation error: [{0:?}]")]
     RequestValidationError(RpcRequestValidationErrorKind),
+    /// An error returned when the server cannot process a request from an rpc client.
     #[error("handler error: [{0}]")]
     HandlerError(E),
+    /// An error returned when the rpc server returns an internal server error.
     #[error("internal error: [{info:?}]")]
     InternalError { info: Option<String> },
+    /// An error returned when the rpc server returns a response without context i.e. a response the client doesn't understand.
     #[error("error response lacks context: {0}")]
     NonContextualError(RpcError),
+    /// Meta errors returned by the rpc server.
     #[error(transparent)]
     ResponseStatusError(JsonRpcServerResponseStatusError),
 }
 
+/// Errors returned by the rpc client.
 #[derive(Debug, Error)]
 pub enum JsonRpcError<E> {
+    /// These errors are returned when there's a problem sending a request to an rpc server.
     #[error(transparent)]
     TransportError(RpcTransportError),
+    /// These errors are returned whenerver there's a problem with the rpc server.
```

Comment:
Problem not necessarily "with" the server. But at least, returned "from" the server.